src/System Application/App/AppSource Gallery/src/AppSourceProductDetails.Page.al

```diff
@@ -0,0 +1,395 @@
+        CurrentRecordCanBeUninstalled := false;
+        CurrentRecordCanBeInstalled := false;
+        if (AppID <> '') then
+            CurrentRecordCanBeUninstalled := ExtensionManagement.IsInstalledByAppID(AppID);
+
+        if ProductObject.Get('plans', ProductPlansToken) then
+            RenderPlans(ProductPlansToken);
+    end;
+
+    procedure RenderPlans(PlansObject: JsonToken)
+    var
+        AllPlans: JsonArray;
+        PlanLinesBuilder: TextBuilder;
+        PlanItem: JsonToken;
+        PlanItemObject: JsonObject;
+        PlanItemArray: JsonArray;
+        MonthlyPriceText, YearlyPriceText : Text;
+        i, availabilitiesAdded : Integer;
+    begin
+        availabilitiesAdded := 0;
+        PlanLinesBuilder.Clear();
+
+        AllPlans := PlansObject.AsArray();
+        for i := 0 to AllPlans.Count() do
+            if AllPlans.Get(i, PlanItem) then begin
+                PlanItemObject := PlanItem.AsObject();
+                if PlanItem.SelectToken('availabilities', PlanItem) then begin
+                    PlanItemArray := PlanItem.AsArray();
+                    if PlanItemArray.Count() > 0 then begin
+                        if BuildPlanPriceText(PlanItemArray, MonthlyPriceText, YearlyPriceText) then
+                            availabilitiesAdded += 1;
+                        PlanLinesBuilder.Append(
+                            StrSubstNo(
+                                PlanLineItemTemplateLbl,
+                                GetStringValue(PlanItemObject, 'displayName'),
+                                GetStringValue(PlanItemObject, 'description'),
+                                MonthlyPriceText,
+                                YearlyPriceText));
+                    end;
+                end;
+            end;
+
+        if (availabilitiesAdded > 0) then begin
+            PlansAreVisible := true;
+            PlansOverview := StrSubstNo(
+                PlanLinesTemplateLbl,
+                PlanLinesColumnPlansLbl,
+                PlanLinesColumnDescriptionLbl,
+                PlanLinesColumnMonthlyPriceLbl,
+                PlanLinesColumnAnnualPriceLbl,
+                PlanLinesBuilder.ToText());
+        end else begin
+            PlansAreVisible := false;
+            PlansOverview := '';
+        end;
+
+        CurrentRecordCanBeInstalled := (AppID <> '') and (not CurrentRecordCanBeUninstalled) and AppSourceProductManager.CanInstallProductWithPlans(AllPlans);
+    end;
+
+    local procedure BuildPlanPriceText(Availabilities: JsonArray; var MonthlyPriceText: Text; var YearlyPriceText: Text): Boolean
+    var
+        item: JsonToken;
+        itemObject: JsonObject;
+        item2: JsonToken;
+        arrayItem: JsonArray;
+        i: integer;
+        currency: text;
+        monthly, yearly : decimal;
+        freeTrial: Boolean;
+    begin
+        freeTrial := false;
+        for i := 0 to Availabilities.Count do
+            if (Availabilities.Get(i, item)) then begin
```

Comment:
No parentheses for single conditions needed (repeated a few time more down below)